### PR TITLE
Extract common parts for L4 LoadBalancers - Part 3

### PR DESCRIPTION
**Pull request by kl52752**
Extract common L4 LB firewall functions

### CODE REVIEW

File: pkg/firewalls/firewalls_l4.go [hunk 1]
```diff
@@ -18,60 +18,75 @@ package firewalls
...
+	"k8s.io/client-go/tools/record"
 	"k8s.io/ingress-gce/pkg/utils"
 	"k8s.io/klog"
 	"k8s.io/legacy-cloud-providers/gce"
 )
 
-func EnsureL4FirewallRule(cloud *gce.Cloud, fwName, lbIP, nsName string, sourceRanges, portRanges, nodeNames []string, proto string, sharedRule bool, l4Type utils.L4LBType) error {
-	existingFw, err := cloud.GetFirewall(fwName)
+// FirewallParams holds all data needed to create firewall for L4 LB
+type FirewallParams struct {
```
Comment (freehan) [inline pkg/firewalls/firewalls_l4.go:33]: I am wondering if it is necessary to expose the unused Firewall fields. For instance, DestRanges etc...

**Review by freehan — COMMENTED** [review]
LGTM overall.

@prameshj wait for second pair of eyes

File: pkg/firewalls/firewalls_l4.go (hunk 1 above)
Comment (kl52752) [reply pkg/firewalls/firewalls_l4.go:33]: Yes you are right, removed.

**Pull request merged**